app/scripts/modules/core/serverGroup/details/serverGroupWarningMessage.service.spec.ts

```diff
@@ -0,0 +1,156 @@
+import {mock} from 'angular';
+import {
+  ServerGroupWarningMessageService,
+  SERVER_GROUP_WARNING_MESSAGE_SERVICE
+} from './serverGroupWarningMessage.service';
+import {ApplicationModelBuilder, APPLICATION_MODEL_BUILDER} from 'core/application/applicationModel.builder';
+import {ServerGroup} from 'core/domain/serverGroup';
+import {Application} from 'core/application/application.model';
+import {IConfirmationModalParams} from 'core/confirmationModal/confirmationModal.service';
+
+describe('serverGroupWarningMessageService', () => {
+  let service: ServerGroupWarningMessageService,
+      applicationModelBuilder: ApplicationModelBuilder,
+      app: Application,
+      serverGroup: ServerGroup;
+
+  beforeEach(mock.module(SERVER_GROUP_WARNING_MESSAGE_SERVICE, APPLICATION_MODEL_BUILDER));
+
+  beforeEach(mock.inject((serverGroupWarningMessageService: ServerGroupWarningMessageService,
+                          _applicationModelBuilder_: ApplicationModelBuilder) => {
+    service = serverGroupWarningMessageService;
+    applicationModelBuilder = _applicationModelBuilder_;
+    app = applicationModelBuilder.createApplication();
+  }));
+
+  describe('addDestroyWarningMessage', () => {
+    it('leaves parameters unchanged when additional server groups exist in cluster', () => {
+      serverGroup = {
+        account: 'test',
+        cloudProvider: 'aws',
+        cluster: 'foo',
+        instanceCounts: { up: 0, down: 0, succeeded: 0, failed: 0, unknown: 0, outOfService: 0 },
+        instances: [],
+        name: 'foo-v000',
+        region: 'us-east-1',
+        type: 'a'
+      };
+
+        app['clusters'] = [
```

Comment:
If we can get https://github.com/spinnaker/deck/pull/3229 merged before this PR, I'll update this (and the other spots using the bracket notation) to `app.clusters`